kevm_pyk/solc_to_k.py

```diff
@@ -0,0 +1,289 @@
+    storage_sentences = _extract_storage_sentences(contract_name, storage_sort, storage_layout)
+    storage_productions, storage_rules = map(list, zip(*storage_sentences))
+    storage_location_production = KProduction([KNonTerminal(contract_sort), KTerminal('.'), KNonTerminal(storage_sort)], KSort('Int'), att=KAtt({'klabel': f'storage_{contract_name}', 'macro': ''}))
+
+    function_sort = KSort(f'{contract_name}Function')
+    function_sentences = _extract_function_sentences(contract_name, function_sort, abi)
+    function_productions, function_rules = map(list, zip(*function_sentences))
+    function_call_data_production = KProduction([KNonTerminal(contract_sort), KTerminal('.'), KNonTerminal(function_sort)], KSort('ByteArray'), att=KAtt({'klabel': f'function_{contract_name}', 'macro': ''}))
+
+    unknownBoolProduction = KProduction([KTerminal('unknownBool')], KSort('Bool'), att=KAtt({'klabel': 'unknownBool', 'symbol': '', 'no-evaluators': '', 'function': '', 'functional': ''}))
+
+    binRuntimeProduction = KProduction([KTerminal('#binRuntime'), KTerminal('('), KNonTerminal(contract_sort), KTerminal(')')], KSort('ByteArray'), att=KAtt({'klabel': 'binRuntime', 'macro': ''}))
+
+    contractProduction = KProduction([KTerminal(contract_name)], contract_sort, att=KAtt({'klabel': f'contract_{contract_name}'}))
+    contractMacro = KRule(KRewrite(KApply('binRuntime', [KConstant(contract_name)]), _parseByteStack(_stringToken(bin_runtime))))
+
+    binRuntimeModuleName = contract_name.upper() + '-BIN-RUNTIME'
+    binRuntimeModule = KFlatModule(
+            binRuntimeModuleName,
+            [KImport('EDSL', True)],
+            [unknownBoolProduction, contractProduction] +
+            storage_productions +
+            [storage_location_production] +
+            storage_rules +
+            function_productions +
+            [function_call_data_production] +
+            function_rules +
+            [binRuntimeProduction, contractMacro]
+        )
+    binRuntimeDefinition = KDefinition(binRuntimeModuleName, [binRuntimeModule], requires=[KRequire('edsl.md')])
+
+    kevm.symbolTable['hashedLocation'] = lambda lang, base, offset: f'#hashedLocation({lang}, {base}, {offset})'
+    kevm.symbolTable['abiCallData'] = lambda fname, *args: f'#abiCallData({fname}{"".join([", " + arg for arg in args])})'
+    kevm.symbolTable['address'] = _typed_arg_unparser('address')
+    kevm.symbolTable['bytes32'] = _typed_arg_unparser('bytes32')
+    kevm.symbolTable['int256'] = _typed_arg_unparser('int256')
+    kevm.symbolTable['uint256'] = _typed_arg_unparser('uint256')
+    kevm.symbolTable['rangeAddress'] = lambda t: f'#rangeAddress({t})'
+    kevm.symbolTable['rangeUInt'] = lambda n, t: f'#rangeUInt({n, t})'
+    kevm.symbolTable['rangeSInt'] = lambda n, t: f'#rangeSInt({n, t})'
+    kevm.symbolTable['binRuntime'] = lambda s: '#binRuntime(' + s + ')'
+    kevm.symbolTable[contract_name] = lambda: contract_name
```

Comment:
Line up the `=` signs for scanability.

Reply:
Done.

Note that this formatting is against the Python [style guide](https://www.python.org/dev/peps/pep-0008/#pet-peeves). I suggest we start a discussion within the team for our own coding guidelines.

Reply:
Hmmm, I guess we need to either adhere to their styles or pick our own style guide I guess.

I usually prefer readability and scanability to anything else, but there is something to be said for adhering to a global style guide. I find it much easier to navigate a file which is visually "pleasing" and structured.

What is your opinion on it?

Reply:
In my opinion, the simplest solution would be to adhere to the official style guide. This way it is easy to enforce conformance using style checkers ([pycodestyle](https://github.com/PyCQA/pycodestyle)) and code formatters (e.g. [black](https://github.com/psf/black) and [isort](https://github.com/PyCQA/isort)) that we can integrate in the CI pipeline. (We might be able to configure the style checker by disabling certain rules, but I think the same approach wouldn't work with the code formatter.)

But on the other hand, as you pointed out, the standard formatting might not be optimal w.r.t. structuring or aesthetics.

Reply:
Let's stick to the existing style actually, because it mirrors what is in upstream pyk library better, and migrating that whole library over will be a pain.

I personally don't care much for linters or style formatters, I believe aesthetics are more important (code is read more often than it's written).